src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -3272,6 +3311,10 @@ void InterpreterSelectQuery::initSettings()
         context->setSetting("group_by_two_level_threshold_bytes", Field(0));
 
     }
+
+    if (query.emit())
+        /// Setup request query mode as streaming
+        context->setSetting("allow_experimental_streaming_query_mode", Field("streaming"));
```

Comment:
For 2) yes, it is questionable for ClickHouse users. In T+ Proton, every query is stream query by default. May I suggest enhancing the modes in another PR since this one is complex / big enough. And we can also collect more feedbacks from the community and revise from there.